share/cmake/utils/CompilerFlags.cmake

```diff
@@ -72,6 +68,14 @@ endif()
 mark_as_advanced(PLATFORM_COMPILE_FLAGS)
 
 
+###############################################################################
+# Define hidden symbol visibility by default for all targets.
+
+set(CMAKE_C_VISIBILITY_PRESET hidden)
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN YES)
```

Comment:
Should these be options/cache variables so that they show up in ccmake, etc. as settable, documented build options?

Reply:
I guess we should prefer options for modern cmake. Do you see a scenario that would justify these being exposed ?

Reply:
Not necessarily. I think you could make a good argument that the symbols should always be hidden, and so it should not be exposed as cache variables. People could still override it with `-D`, but they'd need to read the cmake to find out that it existed; by not being declared as cache variables, they are much less prominently advertised to anybody not specifically looking for them.

Reply:
Given that these are builtin CMake variable and not OCIO specific, I don't think there is much value aliasing them as an option or cache variable. We can probably expect developer wanting to play with these to know them, as it's probably going to be required (if ever) only in very specific / debug tests ?

One thing that I just realised though is that we should probably check that the user didn't override these through the command line or other ways, so that there is still opportunity to change the settings if need be. Maybe that was part of you concerns.

Reply:
I think that OpenColorIO `cmake` should not have dedicated OCIO options (`ccamke` displays nothing related to symbol visibility) because most of people will/should not change the default behavior. However, expert users should have a way to change default values if needed i.e. lines 74 to 76 could be something like:

```
if(CMAKE_C_VISIBILITY_PRESET NOT DEFINED)
  set(CMAKE_C_VISIBILITY_PRESET hidden)
else()
   message(WARNING 'The .... is ${CMAKE_C_VISIBILITY_PRESET}') 
endif()
...
```

Reply:
Implemented just what you described @hodoulp, thanks for the snippet.